service/src/java/org/apache/hive/service/server/HiveServer2.java

```diff
@@ -325,6 +327,11 @@ public synchronized void init(HiveConf hiveConf) {
       scheduledQueryService = ScheduledQueryExecutionService.startScheduledQueryExecutorService(hiveConf);
     }
 
+    if (hiveConf.getBoolVar(ConfVars.HIVE_QUERY_HISTORY_ENABLED)) {
+      queryHistoryService = new QueryHistoryService(hiveConf, serviceContext).start();
```

Comment:
queryHistoryService.start();

Reply:
sorry, I don't get this, what's your point here?
there is a constructor and a fluent start call, that starts the actual "running" things inside